scripts/squash_migrations.sh

```
@@ -15,21 +20,49 @@ if ! command -v atlas &> /dev/null; then
 if [ ! -d "$MIGRATION_DIR" ]; then
     echo "Error: Migration directory not found: $MIGRATION_DIR"
     exit 1
 fi
 
-count=$(find "$MIGRATION_DIR" -maxdepth 1 -name '*.sql' -printf '.' | wc -c)
-echo "Migration count: $count (threshold: $THRESHOLD)"
+ALL_MIGRATIONS=()
+for f in "$MIGRATION_DIR"/*.sql; do
+    [ -f "$f" ] && ALL_MIGRATIONS+=("$(basename "$f")")
+done
+IFS=$'\n' ALL_MIGRATIONS=($(printf '%s\n' "${ALL_MIGRATIONS[@]}" | sort)); unset IFS
```

Shellcheck (0.11.0)

[warning] Prefer mapfile or read -a to split command output (or quote to avoid splitting).

(SC2207)